.github/workflows/publish-to-npm.yml

```diff
@@ -1,21 +1,71 @@
 name: publish-to-npm
 on:
-    pull_request:
-        types: [closed]
-        branches:
-            - main
+  pull_request:
+    types: [ closed ]
+    branches:
+      - main
+      - develop
+      - epic/**
 jobs:
-    publish:
-        if: github.repository == 'adobe/aio-cli-plugin-api-mesh'
-        runs-on: ubuntu-latest
-        steps:
-            - uses: actions/checkout@v4
-            - uses: actions/setup-node@v3
-              with:
-                  node-version: 18
-            - run: yarn install --frozen-lockfile
-            - uses: JS-DevTools/npm-publish@v1
-              with:
-                  token: ${{ secrets.ADOBE_BOT_NPM_TOKEN }}
-                  access: 'public'
-                  tag: 'alpha'
+  publish:
+    if: github.event.pull_request.merged == true
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v4
+
+      - name: Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+
+      - name: Yarn install
+        run: yarn install --frozen-lockfile
+
+      - name: Get version from package.json
+        id: get_version
+        run: | 
+          version=$(jq -r '.version' package.json)
+          echo "version=$version" >> $GITHUB_OUTPUT
+          echo "Read version $version from package.json"
+
+      - name: Verify version corresponds to branch
+        id: verify_version
+        run: |
+          target_branch="${GITHUB_REF#refs/heads/}"
+          version_tag=""
+          
+          if [ "$target_branch" == "main" ]; then
+            version_tag="latest"
+          
+          elif [ "$target_branch" == "develop" ]; then
+            if [[ "${{ steps.get_version.outputs.version }}" =~ beta ]]; then
```

Comment:
I am new to this, so can you confirm if this is checking for `beta` in the version string from `get_version` step?

Reply:
I dont think we should move forward with publishing if the version string does not have `beta` in case of merge to `develop` and `alpha` in case of `epic/*` branches. 

During a publish the tag should match the version

Reply:
I believe this checks if the version is `beta` if merged to develop

Reply:
Yeah, that's the intention here. If it's the `develop` branch, it will only publish if the version has `beta` in it. Additionally, it will append the beta _tag_ to the publish.

If you don't want to publish, then don't make it a `beta` version.